File: 1-calorie-hunting/1-calorie-hunting.py
allCollectedFoodItems = open("input.txt", "r")

bagOfFood = 0
totalForEachElf = []
for foodItem in allCollectedFoodItems:
    if foodItem == "\n":
        totalForEachElf.append(bagOfFood)
        bagOfFood = 0
        continue

    bagOfFood += int(foodItem)

print(max(totalForEachElf))
print(sum(sorted(totalForEachElf)[-4:-1]))

File: 1-calorie-hunting/1-calorie-hunting.js
const { readFileSync } = require('fs')

const elfFoodCollection = readFileSync('./input.txt', 'utf-8').split('\n')

let elfIndex = 0
const elfTotalArray = elfFoodCollection.reduce((acc, cur) => {
    if (!acc[elfIndex]) acc[elfIndex] = 0
    cur === '' ? elfIndex++ : (acc[elfIndex] += +cur)
    return acc
}, [])

console.log('The largest calorie collection is: ' + Math.max(...elfTotalArray))
console.log(
    'The biggest boys collected a total of ' +
        elfTotalArray
            .sort()
            .slice(-4)
            .reduce((acc, cur) => acc + cur, 0) +
        ' calories.'
)
